test/unit/host_test.rb

```diff
@@ -266,6 +266,15 @@ def teardown
     refute host.primary_interface.managed?
   end
 
+  test "#configuration? returns true when host has puppetmaster" do
+    host = FactoryGirl.create(:host)
```

Comment:
do we have to save the record? I think `.build` should be enough here and for smart proxy as well (which will save some time)